pulsar-client/src/main/java/org/apache/pulsar/client/impl/PatternMultiTopicsConsumerImpl.java

```diff
@@ -105,7 +108,60 @@ public void run(Timeout timeout) throws Exception {
         if (timeout.isCancelled()) {
             return;
         }
-        client.getLookup().getTopicsUnderNamespace(namespaceName, subscriptionMode, topicsPattern.pattern(), topicsHash)
+        asyncRecheckTopicsChange().exceptionally(ex -> {
+            log.warn("[{}] Failed to recheck topics change: {}", topic, ex.getMessage());
+            return null;
+        }).thenAccept(__ -> {
+            // schedule the next re-check task
+            this.recheckPatternTimeout = client.timer()
+                    .newTimeout(PatternMultiTopicsConsumerImpl.this,
+                    Math.max(1, conf.getPatternAutoDiscoveryPeriod()), TimeUnit.SECONDS);
+        });
+    }
+
+    private void recheckTopicsChangeRetryIfFailed() {
```

Comment:
`recheckTopicsChangeRetryIfFailed`  `retryRecheckPatternTask`

Reply:
Renamed to `recheckTopicsChangeAfterReconnect`